### PR TITLE
Remove unused "$className" argument

### PR DESCRIPTION
**Pull request by malukenho**
/cc @Ocramius @lcobucci

### CODE REVIEW

File: lib/Doctrine/ORM/Persisters/Entity/AbstractEntityInheritancePersister.php [hunk 1]
```diff
@@ -82,12 +82,11 @@ protected function getSelectColumnSQL($field, ClassMetadata $class, $alias = 'r'
     /**
      * @param string $tableAlias
      * @param string $joinColumnName
-     * @param string $className
      * @param string $type
      *
      * @return string
      */
-    protected function getSelectJoinColumnSQL($tableAlias, $joinColumnName, $className, $type)
+    protected function getSelectJoinColumnSQL($tableAlias, $joinColumnName, $type)
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Persisters/Entity/AbstractEntityInheritancePersister.php]: This is a minor BC break, and needs to be documented in [`UPGRADE.md`](https://github.com/doctrine/doctrine2/blob/master/UPGRADE.md)

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by malukenho** [comment]
@Ocramius now that I looked at it more deeply, this change can break code on users' side as `AbstractEntityInheritancePersister#getSelectJoinColumnSQL` is protected and can be overwrited.

Still valid or can be closed?

File: UPGRADE.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Upgrade to 2.5
 
+## Minor BC BREAK: remove $className parameter on AbstractEntityInheritancePersister#getSelectJoinColumnSQL
```
Comment (Ocramius) [inline UPGRADE.md]: `removed`

Comment (Ocramius) [inline UPGRADE.md]: Put backticks around `AbstractEntityInheritancePersister#getSelectJoinColumnSQL`

File: UPGRADE.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
...
+
+As `$className` parameter is not used on its method body, it was safely removed.
```
Comment (Ocramius) [inline UPGRADE.md]: s/`is not used on its method body`/`was not used in the method`

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
> @Ocramius now that I looked at it more deeply, this change can break code on users' side as `AbstractEntityInheritancePersister#getSelectJoinColumnSQL` is protected and can be overwrited.

Persisters are internal API, and we don't support custom userland persisters anyway. This is still valid.

**Pull request merged**